src/main/java/fi/okm/jod/yksilo/entity/Yksilo.java

```diff
@@ -70,16 +59,20 @@ public class Yksilo {
   @OneToMany(mappedBy = "yksilo", fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
   private Set<Toiminto> toiminnot;
 
-  @OneToMany(fetch = FetchType.LAZY)
+  @ManyToMany(fetch = FetchType.LAZY, mappedBy = "kiinnostuneet")
```

Comment:
Incorrect use of mappedby, the association needs to be unidirectional (Yksilo->Osaaminen)